clang/lib/Basic/Targets/ARM.cpp

```diff
@@ -332,7 +333,7 @@ ARMTargetInfo::ARMTargetInfo(const llvm::Triple &Triple,
     default:
       if (IsNetBSD)
         setABI("apcs-gnu");
-      else if (IsFreeBSD || IsOpenBSD || IsHaiku || IsOHOS)
+      else if (IsFreeBSD || IsFuchsia || IsOpenBSD || IsHaiku || IsOHOS)
```

Comment:
I haven't actually been able to discern how `aapcs-linux` actually differs from `aapcs` or `aapcs-vfp`.

Reply:
The resources I found when researching this is https://wiki.debian.org/ArmEabiPort and https://kanj.github.io/elfs/book/armMusl/cross-tools/abi.html and according to those the only difference is `enum` size: `aapcs` defines `enum`s to be a variable sized type, while with `aapcs-linux` they are always ints (4 bytes).

Reply:
Ah right, that is the default for `-fshort-enum` IOW. I don't know why Linux long ago diverged from the clearly-published AAPCS on this, but for us matching Linux ABIs for this sort of detail (things likely relevant to porting code from arm-linux distinct from direct OS dependencies) is more important than published standards or CPU vendor's recommendations.  It would be nice if the code made it easier to discover that `aapcs-linux` means `aapcs` + `-fshort-enum`.

Reply:
It's especially notable if not only the glibc-based Linux configurations, but musl, android, etc. are also still doing `-fshort-enum` / `aapcs-linux` then there is no question that Fuchsia wants to follow them all.  If e.g. Android differed from `-gnueabihf` on this then we might consider whether to align with one or the other.